### DISCUSSION TITLE
Delay of 3 seconds in ride request popup for all drivers to accept the ride

### DISCUSSION THREAD

**Discussion by hkmangla** [nammayatri/nammayatri#307] [Ideas]
We have added a delay of 3 seconds in all ride request popup of drivers so that they will have time to check the origin and destination locations of the ride. Along with that they can also verify how far is the pickup point from their location. 
It can help reducing the cancellation as sometime drivers are accepting without checking the origin and destination then cancelling later on.

**Comment by Magizhan** [comment]
Thanks @hkmangla . While it feels logical that this wait will help drivers take an informed decision of accepting a ride that works for them and also help in reducing cancellations, there was also a feedback from one of the driver over call. He said 

> Many drivers are disappointed because of this change and want to accept as fast as possible. Ola doesn't have it.  

I did tell him that this would help in reducing the cancellations because the drivers get time to read for 3s and then accept and not accept and then cancel. However, he was stressing that 

> Uber is the best (the ride request is sent one by one to the drivers and not broadcasted) , next is Ola (no wait time) and not the Rapido way (5s wait time). Some drivers might find it better but many don't. We still have to understand this better. We have also noticed that some drivers have an auto click app to accept the rides immediately.

Let's monitor the data and feedback for the next few days. Should we also notify drivers over in-app chat about this change and get their feedback through comments?

**Reply by hkmangla** [reply]
Sure @Magizhan . We can monitor the data and also send an alert to all drivers about the change, so that they can provide their feedback.

**Reply by Magizhan** [reply]
Another feedback from one of the top drivers. Checking the reason. Will update once available. 

<img width="308" alt="Screen Shot 2023-03-25 at 10 49 32 AM" src="https://user-images.githubusercontent.com/34025961/227697945-ed91e082-061f-49f9-a38c-0695be81978c.png">

**Reply by Magizhan** [reply]
Ayaz Ullah says he doesn't look at any of the details before accepting the ride and he doesn't cancel too and so doesn't want this.

**Reply by Magizhan** [reply]
Got a call from Satya on feedback from Rudramurthy and few others that this is perceived as 
1. selective timings for drivers as a penalty and the reason for penalty is unknown
2. I look at the options and carefully choose & I'm fine if the option is pulled off before I do that
3. This is not a favorite option with the incumbents. 
Rather, to solve the problem of accepting first and cancelling later, Satya suggests to mention about the missed opportunities and cancellations in the earnings/performance section. 

@hkmangla , Prithvi Do we see any significant reduction in driver cancellations? If not, we can reduce the default wait to zero as there is unilateral feedback on not wanting this default wait.

**Reply by Prithivirajkota** [reply]
@Magizhan There is a clear improvement in driver cancellation compared to previous 5 days by around 3%. 
![Screenshot 2023-03-25 at 12 33 20 PM](https://user-images.githubusercontent.com/128886402/227702531-86193255-5a48-4512-8b0d-556b5a2c40a6.png)

**Reply by Magizhan** [reply]
@Prithivirajkota Isn't there a delay that can be configured for those who have high cancellation rate? Would we get a similar result even if we measure and reduce that threshold?

**Reply by Magizhan** [reply]
And @Prithivirajkota Can you also communicate in-app about this to all the drivers and request to share concerns if they have any?

**Reply by Prithivirajkota** [reply]
@Magizhan we do have a delay of 5 secs for drivers with atleast 5 bookings and having cancellation rate >40% in a rolling window of 7 days.

**Reply by Prithivirajkota** [reply]
Sure, we can communicate to drivers reg. the 3 sec delay and ask for their concerns.

**Comment by hkmangla** [comment]
After getting too many calls from our champion drivers to remove the default pop delay of 3 seconds. We are reverting back this change

Announcement for the same: #318
